src/core/features-container.h

```diff
@@ -0,0 +1,43 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+#pragma once
+
+#include <src/core/has-features-interface.h>
+
+
+namespace librealsense {
+
+
+class features_container : public virtual has_features_interface
+{
+public:
+    bool supports_feature( feature_id id, std::shared_ptr< feature_interface > * feat_obj ) override
+    {
+        auto it = _features.find( id );
+        if( it != _features.end() )
+        {
+            if( feat_obj != nullptr )
+                feat_obj = &it->second;
```

Comment:
I think you want `*feat_obj = it->second`

Reply:
Fixed